internal/forest/namespaceobjects.go

```diff
@@ -28,12 +29,26 @@ func (ns *Namespace) HasSourceObject(gvk schema.GroupVersionKind, oo string) boo
 }
 
 // DeleteSourceObject deletes a source object by name.
-func (ns *Namespace) DeleteSourceObject(gvk schema.GroupVersionKind, nm string) {
+func (ns *Namespace) DeleteSourceObject(log logr.Logger, gvk schema.GroupVersionKind, nm string) {
+
+	log.Info("Deleting source object in namespace", "namespace", ns.Name(), "gvk", gvk, "nm", nm)
 	delete(ns.sourceObjects[gvk], nm)
 	// Garbage collection
 	if len(ns.sourceObjects[gvk]) == 0 {
 		delete(ns.sourceObjects, gvk)
 	}
+
```

Comment:
add an annotation which prevents any resource hnc was not aware of being deleted.